pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerStats.java

```diff
@@ -114,4 +114,9 @@ public interface ConsumerStats extends Serializable {
      * @return
      */
     Map<Long, Integer> getMsgNumInSubReceiverQueue();
+
+    /**
+     * @return stats for each partition if topic is partitioned topic
+     */
+    Map<String, ConsumerStats> getPartitionStats();
```

Comment:
Here is not only about partition stats. Users can provide topic lists and topic patterns.
Is it better to use a new interface `MultiTopicConsumerStats extends ConsumerStats`?

For the producer side, we can have a new interface `PartitionedTopicProducerStats extends ProducerStats`

So that we can add new methods to the `MultiTopicConsumerStats` and `PartitionedTopicProducerStats` such `getStatsForInternalTopics` and `getStatsForPartitions`

Reply:
yes, that's a good idea to make partitioned stats more abstract and extendable.

Reply:
yes, sure, we can have separate interface for partitioned topic with more clear code. 
However, the current implementation with combined stats-recorder and stats-interface implementation has complicated things though they are two separate entities (recorder can't become stats but recorder should record the stats ). combining both entities incorrectly complicates the inheritance.
We also should still keep the Partitioned topic abstraction from user. Producer/Consumer::getStats() should return one interface with all required API and user doesn't have to cast based on topic type. Because Partitioned-topic-stats are just a collection of stats so, we should have getPartitionedTopicStats() API with in Stats interface and it should return empty response if topic is not partitioned topic.
Atleast this will be a good start to creating the abstraction and without having existing API changes in this PR.
I will create a separate PR with further improvement.

This PR has addressed all comments.